src/main/java/net/neoforged/neoforge/flag/FlagManager.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.flag;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Stream;
+import net.neoforged.neoforge.common.extensions.ILevelReaderExtension;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * The main manager for all {@link Flag} related tasks.
+ * <p>
+ * An instance of this can be looked up via {@link ILevelReaderExtension#getModdedFlagManager()}.
+ */
+public sealed interface FlagManager permits DummyFlagManager, NullFlagManager, ServerFlagManager {
+    /**
+     * Null instance to be used when no other can be found.
+     */
+    FlagManager NULL = new NullFlagManager();
```

Comment:
This should be `FlagManager.empty()` which returns a `EmptyFlagManager.INSTANCE` or some such.